File: app/src/main/java/org/michaelbel/moviemade/rest/api/PEOPLE.java
package org.michaelbel.moviemade.rest.api;

import org.michaelbel.moviemade.rest.model.Person;
import org.michaelbel.moviemade.rest.response.MoviePeopleResponse;
import org.michaelbel.moviemade.rest.response.PeopleResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PEOPLE {

    @GET("person/{person_id}?")
    Observable<Person> getDetails(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language,
        @Query("append_to_response") String response
    );

    @GET("person/{person_id}/movie_credits?")
    Observable<MoviePeopleResponse> getMovieCredits(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("person/{person_id}/tv_credits?")
    Observable<?> getTVCredits(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("person/{person_id}/combined_credits?")
    Observable<?> getCombinedCredits(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("person/{person_id}/external_ids?")
    Observable<?> getExternalIDs(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("person/{person_id}/images?")
    Observable<?> getImages(
        @Path("person_id") int id,
        @Query("api_key") String apiKey
    );

    @GET("person/{person_id}/tagged_images?")
    Observable<?> getTaggedImages(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("page") int page
    );

    @GET("person/{person_id}/changes?")
    Observable<?> getChanges(
        @Path("person_id") int id,
        @Query("api_key") String apiKey,
        @Query("end_date") String endDate,
        @Query("start_date") String startDate,
        @Query("page") int page
    );

    @GET("person/latest?")
    Observable<?> getLatest(
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("person/popular?")
    Observable<PeopleResponse> getPopular(
        @Query("api_key") String apiKey,
        @Query("language") String language,
        @Query("page") int page
    );
}

File: app/src/main/java/org/michaelbel/moviemade/data/service/PEOPLE.kt
package org.michaelbel.moviemade.data.service

import io.reactivex.Observable
import org.michaelbel.moviemade.data.dao.CreditsResponse
import org.michaelbel.moviemade.data.dao.Person
import org.michaelbel.moviemade.data.dao.PersonsResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface PEOPLE {

    @GET("person/{person_id}?")
    fun getDetails(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("append_to_response") response: String
    ): Observable<Person>

    @GET("person/{person_id}/changes?")
    fun getChanges(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("end_date") endDate: String,
        @Query("start_date") startDate: String,
        @Query("page") page: Int
    ): Observable<*>

    @GET("person/{person_id}/movie_credits?")
    fun getMovieCredits(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<CreditsResponse>

    @GET("person/{person_id}/tv_credits?")
    fun getTVCredits(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<*>

    @GET("person/{person_id}/combined_credits?")
    fun getCombinedCredits(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<*>

    @GET("person/{person_id}/external_ids?")
    fun getExternalIDs(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<*>

    @GET("person/{person_id}/images?")
    fun getImages(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String
    ): Observable<*>

    @GET("person/{person_id}/tagged_images?")
    fun getTaggedImages(
        @Path("person_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("page") page: Int
    ): Observable<*>

    // getTranslations

    @GET("person/latest?")
    fun getLatest(
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<*>

    @GET("person/popular?")
    fun getPopular(
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("page") page: Int
    ): Observable<PersonsResponse>
}